testing-overhead/src/test/java/io/opentelemetry/results/AppPerfResults.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.results;
+
+import io.opentelemetry.agents.Agent;
+import io.opentelemetry.config.TestConfig;
+
+public class AppPerfResults {
+
+  private final Agent agent;
+  private final TestConfig config;
+  private final double iterationAvg;
+  private final double iterationP95;
+  private final double requestAvg;
+  private final double requestP95;
+  private final long totalGCTime;
+  private final long totalAllocated;
+  private final MinMax heapUsed;
+  private final float maxThreadContextSwitchRate;
+  private final long startupDurationMs;
+  private final long peakThreadCount;
```

Comment:
`public final` and remove all those getters?

Reply:
I did use direct field access on the `MinMax` class, but getters seem way more idiomatic. I'm totally fine with direct field access for immutable things, but I thought it would be discouraged. :). What do other folks think?

Reply:
Whatever, as long as it is immutable 😄